sdk/src/test/java/io/dapr/client/DaprHttpTest.java

```diff
@@ -27,6 +29,9 @@
 
 public class DaprHttpTest {
 
+  @Rule
+  public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
```

Comment:
No need for this since you can achieve the override by system properties.

Reply:
Right we could use the system properties but the main workflow we are trying to test is: when a user sets DAPR_API_TOKEN env variable then java-sdk needs to set dapr-api-token header in all dapr requests.

If we just test using system properties and let's say someone changes the behavior of Properties.getValueOrDefault() to not check environment variables then the test would still pass but [this use case](https://github.com/dapr/docs/tree/master/howto/enable-dapr-api-token-based-authentication#self-hosted) would be broken.

Reply:
You are right about the use case test, on the other hand, unit test assume abstractions from other classes. The scenario you are describing would be better tested in an integration test. I am OK with this now mainly because the dependency is for tests only and not released.